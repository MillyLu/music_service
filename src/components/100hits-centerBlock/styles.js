import styled from "styled-components";

export const MainCenterblock = styled.div`
width: auto;
flex-grow: 3;
padding: 20px 40px 20px 111px;
background-color: ${props => props.theme === "light" ? "#FFFFFF" : "#1C1C1C"}; 
`;

export const CenterblockContent = styled.div`
display: flex;
flex-direction: column;
width: 1107px;
`;