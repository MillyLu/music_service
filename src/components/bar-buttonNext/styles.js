import styled from "styled-components";
import {ReactComponent as NextSvg} from '../svg/next/next.svg';

export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonNext = styled(PlayerButton)`
margin-right: 28px;
fill: #a53939;
`;

export const PlayerButtonNextSvg = styled(NextSvg)`
width: 15px;
height: 14px;
fill: ${props => props.theme === "light" ? "#B1B1B1" : "#D9D9D9"};
 & :hover {
    fill: ${props => props.theme === "light" ? "#707070" : "#696969"};
 }
`;