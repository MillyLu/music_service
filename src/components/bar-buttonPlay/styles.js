import styled from "styled-components";

export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const ButtonCursor = styled(PlayerButton)`
cursor: pointer;
`

export const PlayerButtonPlay = styled(ButtonCursor)`
margin-right: 23px;
`;

export const PlayerButtonPlaySvg = styled.svg`
width: 22px;
  height: 20px;
  fill: ${props => props.theme === "light" ? "#B1B1B1" : "#D9D9D9"};
  & :hover {
     fill: ${props => props.theme === "light" ? "#707070" : "#696969"};
  }
`;

export const Audio = styled.audio`
display: none;
`