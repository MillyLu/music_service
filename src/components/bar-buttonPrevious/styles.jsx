import styled from "styled-components";
import {ReactComponent as PrevSvg} from '../svg/prev/prev.svg';

export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonPrev = styled(PlayerButton)`
margin-right: 23px;
`;

export const PlayerButtonPrevSvg = styled(PrevSvg)`
width: 15px;
height: 14px;
fill: ${props => props.theme === "light" ? "#B1B1B1" : "#D9D9D9"};
 & :hover {
    fill: ${props => props.theme === "light" ? "#707070" : "#696969"};
 }
`;