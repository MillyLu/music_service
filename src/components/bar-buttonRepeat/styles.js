import styled from "styled-components";
import {ReactComponent as RepeatSvg} from '../svg/repeat/repeat.svg';


export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonIcon = styled(PlayerButton)``;

export const PlayerButtonRepeat = styled(PlayerButtonIcon)`
margin-right: 24px;
`;

export const PlayerButtonRepeatSvg = styled(RepeatSvg)`
width: 18px;
height: 12px;
fill: none;
stroke: ${props => props.theme === "light" ? "#B1B1B1" : "#696969"};

${PlayerButtonRepeat}:hover & {
    fill: none;
    stroke: ${props => props.theme === "light" ? "#707070" : "#ACACAC"};
    cursor: pointer;
  }
${PlayerButtonRepeat}:active & {
    fill: ${props => props.theme === "light" ? "#000000" : "#FFFFFF"};
    stroke: ${props => props.theme === "light" ? "#000000" : "#FFFFFF"};
    cursor: pointer;
  }
`;