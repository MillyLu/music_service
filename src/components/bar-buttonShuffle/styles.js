import styled from "styled-components";
import {ReactComponent as ShuffleSvg} from '../svg/shuffle/shuffle.svg';

export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonIcon = styled(PlayerButton)``;

export const PlayerButtonShuffle = styled(PlayerButtonIcon)`
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonShuffleSvg = styled(ShuffleSvg)`
width: 19px;
height: 12px;
fill: none;
stroke: ${props => props.theme === "light" ? "#B1B1B1" : "#696969"};
${PlayerButtonShuffle}:hover & {
    fill: none;
    stroke: ${props => props.theme === "light" ? "#707070" : "#ACACAC"};
    cursor: pointer;
  }
${PlayerButtonShuffle}:active & {
    fill: ${props => props.theme === "light" ? "#000000" : "#FFFFFF"};
    stroke: ${props => props.theme === "light" ? "#000000" : "#FFFFFF"};
    cursor: pointer;
  }
`;