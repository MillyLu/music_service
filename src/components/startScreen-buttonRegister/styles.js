import styled from "styled-components";


export const StartscreenButton = styled.button`
width: 278px;
height: 52px;
background: #580EA2;
border-radius: 6px;
font-family: 'StratosSkyeng';
font-style: normal;
font-weight: 400;
font-size: 18px;
line-height: 24px;
letter-spacing: -0.003em;
color: #FFFFFF;
display: flex;
justify-content: center;
align-items: center;
margin-top: 20px;
border-radius: 6px;
&:hover{
    background: #3F007D;
}
&:active{
    background: #271A58;
}
`;


export const StartscreenButtonReg = styled(StartscreenButton)`
color: #000000;
border: 1px solid #D0CECE;
border-radius: 6px;
background: white;
border: 1px solid #D0CECE;
&:hover{
    background: #F4F5F6;
}
&:active{
    background: #D9D9D9;
}
`;