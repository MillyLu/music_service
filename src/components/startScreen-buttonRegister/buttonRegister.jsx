import { useNavigate } from "react-router-dom";
import * as Styled from './styles';



export function StartscreenButtonRegister () {
    const navigate = useNavigate();
    const path = "/register";

    const handleClick = (e) => {
        e.preventDefault();
        navigate( path, { replace: true });
      
        }


    return(
        <Styled.StartscreenButtonReg onClick={handleClick}>Зарегистрироваться</Styled.StartscreenButtonReg>
    )
}