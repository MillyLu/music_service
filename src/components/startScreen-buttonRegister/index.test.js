/* eslint-disable react/jsx-no-undef */
/* eslint-disable arrow-body-style */
/* eslint-disable react/function-component-definition */
/* eslint-disable no-undef */
import { render, screen, fireEvent } from "@testing-library/react";
import { StartscreenButtonRegister } from "./buttonRegister";
import * as Styled from './styles';


const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom') ,
  useNavigate: () => mockedUsedNavigate,
}));


describe('<StartscreenButtonRegister>', () => {
    it('should render and call handleClick on click', () => {
        const { container} = render(<StartscreenButtonRegister>Зарегистрироваться</StartscreenButtonRegister>);
        expect(container).toMatchSnapshot();
     });

    it('call handleClick on click', () => {
        const handleClick = jest.fn();


        render(
            <Styled.StartscreenButtonReg onClick={handleClick}>Зарегистрироваться</Styled.StartscreenButtonReg>
          );
   
          fireEvent.click(screen.getByText('Зарегистрироваться'));
   
          expect(handleClick).toBeCalledTimes(1);
    }) 
})