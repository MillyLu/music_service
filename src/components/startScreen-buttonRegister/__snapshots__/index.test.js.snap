// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartscreenButtonRegister> should render and call handleClick on click 1`] = `
<div>
  <button
    class="sc-bcXHqe sc-gswNZR cODikO lldMik"
  >
    Зарегистрироваться
  </button>
</div>
`;
